src/arithmetic.c

```diff
@@ -132,15 +132,13 @@ static const char arithmetic_crystal_class_symbols[74][7] = {
 };
 
 int arth_get_symbol(char symbol[7], const int spgroup_number) {
-    int i, arth_number;
-
     if (spgroup_number < 1 || spgroup_number > 230) {
         return 0;
     }
 
-    arth_number = arithmetic_crystal_classes[spgroup_number];
+    int arth_number = arithmetic_crystal_classes[spgroup_number];
     memcpy(symbol, arithmetic_crystal_class_symbols[arth_number], 7);
-    for (i = 0; i < 6; i++) {
+    for (int i = 0; i < 6; i++) {
```

Comment:
This change moves the initializations near where the variables are used, e.g. only within the for loop.
- This allows the compiler to unroll the loops efficiently
- Also guarantees that the compiled nested functions have their own local variables and are thus more efficient in their memory access

Reply:
It was necessary to support an old standard compiler. Now OK, I believe. If I remember correctly, I wasn't allowed even using `pow` function for that compiler...